From 0cc17f41dc2950dedf3c37b72c94c61a278ae5ca Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Sat, 29 Jun 2019 16:42:50 +0800
Subject: [PATCH] invoke grub-install to generate efi filesystem

Previously, the efi filesystem was from package grub-efi
which generated at build time.

Add a wraper to generate efi filesystem with fs_uuid located.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Replace util.getSysroot() with conf.target.system_root.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 34.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 pyanaconda/modules/storage/bootloader/efi.py | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/pyanaconda/modules/storage/bootloader/efi.py b/pyanaconda/modules/storage/bootloader/efi.py
index 158287b1e..87ae27f2c 100644
--- a/pyanaconda/modules/storage/bootloader/efi.py
+++ b/pyanaconda/modules/storage/bootloader/efi.py
@@ -126,6 +126,23 @@ class EFIBase(object):
             self.remove_efi_boot_target()
         self.add_efi_boot_target()
 
+        with open(conf.target.system_root + "/tmp/grub-mkimage-wrapper.sh", "w") as f:
+            f.write('#!/bin/sh\n')
+            f.write('uuid=`grub-probe  --target=fs_uuid  /boot/efi/`\n')
+            f.write('echo "search.fs_uuid ${uuid} root" > /tmp/load.cfg\n')
+            f.write('echo "grub-mkimage -c /tmp/load.cfg $@ search_fs_uuid"\n')
+            f.write('grub-mkimage -c /tmp/load.cfg $@ search_fs_uuid\n')
+        os.chmod(conf.target.system_root + "/tmp/grub-mkimage-wrapper.sh", 0o755)
+
+        rc = util.execInSysroot("grub-install", ["--removable",
+                                                  "--efi-directory=/boot/efi/",
+                                                  "--boot-directory=%s" % self.config_dir,
+                                                  "--grub-mkimage=/tmp/grub-mkimage-wrapper.sh",
+                                                  "/boot/efi/"
+                                                 ])
+        if rc:
+            raise BootLoaderError("boot loader install failed")
+
 
 class EFIGRUB(EFIBase, GRUB2):
     """EFI GRUBv2"""
-- 
2.7.4

